lib/using_restricted_array.rb

```diff
@@ -6,56 +6,110 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: ? O(n)
+# Space complexity: ? O(1)
 def length(array)
-  raise NotImplementedError
+  counter = 0
+  until array[counter] == nil
+    counter += 1
+  end
+  return counter
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: ? O(n)
+# Space complexity: ? O(1)
 def print_array(array)
-  raise NotImplementedError
+  index = 0
+  until array[index] == nil 
+    print array[index]
+  end
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: ? O(n)
+# Space complexity: ? O(1)
 def search(array, length, value_to_find)
-  raise NotImplementedError
+  array_search_value = false 
+  index = 0
+  
+  until array[index] == nil
+    if array[index] == value_to_find
+      return true
+    end 
+    index += 1   
+  end
+  return array_search_value
 end
 
 # Finds and returns the largest integer value the array
 # Assumes that the array is not sorted.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: ? O(n)
+# Space complexity: ? O(1)
 def find_largest(array, length)
-  raise NotImplementedError
+  biggest_integer = 0
```

Comment:
What if the array has negative numbers?  better to set `biggest_integer = array[0]`